python/shipDet_conf.py

```diff
@@ -1,486 +1,906 @@
-
- for x in detectorList:
-   if x.GetName() in exclusionList: continue
-   run.AddModule(x)
-# return list of detector elements
- detElements = {}
- for x in run.GetListOfModules(): detElements[x.GetName()]=x
- return detElements
+
+def getParameter(x, ship_geo, latestShipGeo):
+    lv = x.split(".")
+    last = lv[len(lv) - 1]
+    top = ""
+    for l in range(len(lv) - 1):
+        top += lv[l]
+        if l < len(lv) - 2:
+            top += "."
+    a = getattr(ship_geo, top)
+    if hasattr(a, last):
+        return getattr(a, last)
+        # not in the list of recorded parameteres. probably added after
+        # creation of file. Check newest geometry_config:
+    a = getattr(latestShipGeo, top)
+    return getattr(a, last)
+
+
+def posHcal(z, hfile, HcalOption):
+    HcalZSize = 0
+    sz = hfile + "z" + str(z) + ".geo"
+    floc = os.environ["FAIRSHIP"] + "/geometry"
+    f_hcal = floc + "/" + hfile
+    f_hcalz = floc + "/" + sz
+    f = open(f_hcal)
+    rewrite = True
+    if sz in os.listdir(floc):
+        test = os.popen("diff " + f_hcal + " " + f_hcalz).read()
+        if str.count(test, "---") == 1 and not test.find("Position") < 0:
+            rewrite = False  # only different is z position
+    if rewrite:
+        fn = open(f_hcalz, "w")
+    for l in f.readlines():
+        if rewrite:
+            if not l.find("ZPos") < 0:
+                l = "ZPos=" + str(z) + "	#Position of Hcal  center	[cm]\n"
+            fn.write(l)
+        if not l.find("HcalZSize") < 0:
+            HcalZSize = float(l[len("HcalZSize") + 1 :].split("#")[0])
+    f.close()
+    if rewrite:
+        fn.close()
+    if HcalOption == 2:
+        hcal = ROOT.hcal("Hcal", ROOT.kFALSE, sz)
+    else:
+        hcal = ROOT.hcal("Hcal", ROOT.kTRUE, sz)
+    return hcal, HcalZSize
+
+
+def makeEcalGeoFile(z, efile):
+    EcalZSize = 0
+    sz = efile + "z" + str(z) + ".geo"
+    floc = os.environ["FAIRSHIP"] + "/geometry"
+    f_ecal = floc + "/" + efile
+    f_ecalz = floc + "/" + sz
+    f = open(f_ecal)
+    rewrite = True
+    if sz in os.listdir(floc):
+        test = os.popen("diff " + f_ecal + " " + f_ecalz).read()
+        if str.count(test, "---") == 1 and not test.find("Position") < 0:
+            rewrite = False  # only different is z position
+    if rewrite:
+        fn = open(f_ecalz, "w")
+    for l in f.readlines():
+        if rewrite:
+            if not l.find("ZPos") < 0:
+                l = "ZPos=" + str(z) + "	#Position of Ecal start		[cm]\n"
+            fn.write(l)
+        if not l.find("EcalZSize") < 0:
+            EcalZSize = float(l[len("EcalZSize") + 1 :].split("#")[0])
+    f.close()
+    if rewrite:
+        fn.close()
+    return EcalZSize, sz
+
+
+def posEcal(z, efile):
+    EcalZSize, sz = makeEcalGeoFile(z, efile)
+    ecal = ROOT.ecal("Ecal", ROOT.kTRUE, sz)
+    return ecal, EcalZSize
+
+def configure_veto(yaml_file):
```

Comment:
We might eventually want to load the yaml directly in the C++ file, and here only for information needed by other systems/globally.
But for now, this is a nice solution that requires no changes elsewhere.